api/pyproject.toml

```diff
@@ -0,0 +1,65 @@
+[project]
+name = "api"
+version = "0.0.0" # redundant for application
+description = "API for the Openverse project"
+readme = "README.md"
+license = {text = "MIT"}
+authors = [
+  {name = "Openverse", email = "openverse@wordpress.org" },
+]
+
+requires-python = "==3.11.*"
+dependencies = [
+  "adrf >=0.1.4, <0.2",
+  "aiohttp >=3.9.3, <4",
+  "aws-requests-auth >=0.4.3, <0.5",
+  "deepdiff >=6.7.1, <7",
+  "django >=4.2.11, <5",
+  "django-asgi-lifespan >=0.2.0, <0.3",
+  "django-cors-headers >=4.3.1, <5",
+  "django-log-request-id >=2.1.0, <3",
+  "django-migrations-git-conflicts >=1.0, <2",
+  "django-oauth-toolkit >=2.3.0, <3",
+  "django-redis >=5.4.0, <6",
+  "django-split-settings >=1.2.0, <2",
+  "django-tqdm >=1.3.1, <2",
+  "django-uuslug >=2.0.0, <3",
+  "djangorestframework >=3.14.0, <4",
+  "drf-spectacular >=0.27.1, <0.28",
+  "elasticsearch >=8.13.0, <9",
+  "elasticsearch-dsl >=8.12.0, <9",
+  "future >=0.18.3, <0.19",
+  "limit >=0.2.3, <0.3",
+  "pillow >=10.2.0, <11",
+  "psycopg >=3.1.18, <4",
+  "python-decouple >=3.8, <4",
+  "python-xmp-toolkit >=2.0.2, <3",
+  "sentry-sdk >=1.40.6, <2",
+  "uvicorn[standard] >=0.27.1, <0.28",
+  "openverse-attribution @ file:///${PROJECT_ROOT}/../py-packages/openverse-attribution"
+]
+
+[tool.pdm]
+distribution = false
+
+[tool.pdm.dev-dependencies]
+# These packages override their main counterparts with an editable installation.
+overrides = [
+  "-e openverse-attribution @ file:///${PROJECT_ROOT}/../py-packages/openverse-attribution"
+]
+dev = [
+  "ipython >=8.22.1, <9",
+  "pgcli >=3.5.0, <4",
+  "remote-pdb >=2.1.0, <3",
+]
+test = [
```

Comment:
Is there a functional difference between `dev` and `test` here, or is the separation only semantic?

Reply:
This allows us to selectively build smaller and smaller images from dev (all deps) to CI (main + test) to prod (main only).

Reply:
Gotcha! I hadn't seen the test install happening anywhere, so I wasn't sure.